server/channels/app/email/service.go

```diff
@@ -162,14 +162,24 @@ type ServiceInterface interface {
 	SendNoCardPaymentFailedEmail(email string, locale string, siteURL string) error
 	SendRemoveExpiredLicenseEmail(ctaText, ctaLink, email, locale, siteURL string) error
 	AddNotificationEmailToBatch(user *model.User, post *model.Post, team *model.Team) *model.AppError
-	GetMessageForNotification(post *model.Post, translateFunc i18n.TranslateFunc) string
+	GetMessageForNotification(post *model.Post, teamName, siteUrl string, translateFunc i18n.TranslateFunc) string
+	GenerateHyperlinkForChannels(postMessage, teamName, teamURL string) (string, error)
 	InitEmailBatching()
 	SendChangeUsernameEmail(newUsername, email, locale, siteURL string) error
 	CreateVerifyEmailToken(userID string, newEmail string) (*model.Token, error)
 	SendIPFiltersChangedEmail(email string, userWhoChangedFilter *model.User, siteURL, portalURL, locale string, isWorkspaceOwner bool) error
+	SetStore(st store.Store)
 	Stop()
 }
 
+func (es *Service) Store() store.Store {
+	return es.store
+}
+
+func (es *Service) SetStore(st store.Store) {
+	es.store = st
+}
+
```

Comment:
Are these two needed? For setting the mock store in the tests, you can call `th.App.Srv().SetStore()` if I'm not mistaken. In the case of `Store()`, external users of the interface should not need to access it, and the places you use it here you can access the internal `store` directly. Users of the interface should not need to access it.

Reply:
In the Service interface we can't access `store` from `server` or `app` as there is no such parameter. Now we are using `es.Store()` at `GenerateHyperlinkForChannels` where `es` is a `Service`. That's why I need to put mock store on the `store` of service.

Reply:
Correct me if I'm mistaken, but `GenerateHyperlinkForChannels` is a method of [this `Service` struct](https://github.com/mattermost/mattermost/blob/9e99280a40f130d4ba708076dcc5b50aff47d07d/server/channels/app/email/service.go#L44-L55). Others methods there like `GetMessageForNotification` are using `es.store` in the current code.

Reply:
Yes, you are correct. Though in the previous implementation of getMessageForNotification we didn't have to call store method.  Now we combine the generateHyperLink process in this method as a refactoring, where we are using store to get teamName and mentioned channel name.

Reply:
@agarciamontoro Also `GetMessageForNotification` was not covered in the unit tests before. Now we have covered this in the unit tests. That's why we need to set the mock store

Reply:
Ah! I had to go through the code one more time, but I get it now! You're absolutely right, we do need it for the test. Thanks for your patience.

Reply:
You're welcome